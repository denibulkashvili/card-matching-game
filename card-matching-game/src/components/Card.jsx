import React, { Component } from 'react'

class Card extends Component {
  
  render() {
    return (
      <div 
        className={`column is-one-quarter single-card has-text-centered ${!this.props.isCardFrontSide ? 'card-back' : ''} `} 
        onClick={!this.props.isCardFrontSide ? () => this.props.handleCardClick(this.props.index) : null} >
        {this.props.isCardFrontSide ? this.props.word : ''}
      </div>
    );
  }
}

export default Card;