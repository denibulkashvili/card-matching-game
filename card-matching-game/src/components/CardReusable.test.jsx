import React from 'react';
import '../tests/setup';
// import { shallow } from 'enzyme';
import Card from './CardReusable';
import renderer from 'react-test-renderer';

describe('<Card /> component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Card />);
    expect(tree).toMatchSnapshot()
  })
});