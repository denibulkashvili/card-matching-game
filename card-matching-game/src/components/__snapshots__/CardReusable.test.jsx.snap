// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> component matches snapshot 1`] = `
<img
  alt=""
  className="image"
/>
`;
